janus_core/helpers/utils.py

```diff
@@ -407,3 +407,26 @@ def track_progress(sequence: Union[Sequence, Iterable], description: str) -> Ite
 
     with progress:
         yield from progress.track(sequence, description=description)
+
+
+def check_files_exist(config: dict, req_file_keys: Sequence[PathLike]) -> None:
```

Comment:
Is this a more generally useful function? In which case we may want to consider the following:

- Do we want this to `raise` or would it be better to return a bool/set of missing and allow functions to determine whether to raise?
- Missing `file_key`s may also be a sign of problems. Does that want a bool to handle them?

Reply:
I can imagine it could be, but at the moment I don't think we do anything quite like this elsewhere at the moment.

Here, and in `train`, we're only really passing the config files on to MACE, and so this is just an extra protection to handle errors nicely. From what I've seen, missing keys are handled clearly enough by their CLI, so I don't think it's needed yet.

In most other cases, the config is completely optional, and we (should) already have relevant error handling dealt with, either in the Python calculation (e.g. missing restart files for MD), or by typer (e.g. `--struct` being required).

I think for now it's probably fine, until we find uses for it elsewhere, but it's definitely useful to consider.